src/components/useIntersectionTracker/useIntersectionTracker.ts

```diff
@@ -0,0 +1,33 @@
+import { MutableRefObject, RefObject, useEffect } from "react";
```

Comment:
set up a new folder `src/hooks` .  Move the `useIntersectionTracker` implementation into that folder. That folder will contain all project hooks.